lightningd/channel.c

```diff
@@ -154,14 +161,78 @@ struct channel *new_channel(const tal_t *ctx,
 	return channel;
 }
 
-/* FIXME: We could cache this. */
-struct bitcoin_tx *channel_tx(const tal_t *ctx,
-			      const struct channel *channel,
-			      const struct pubkey *per_commitment_point,
-			      const struct htlc ***htlcmap,
-			      enum side side)
+static void add_htlcs(struct bitcoin_tx ***txs,
+		      const u8 ***wscripts,
+		      const struct htlc **htlcmap,
+		      const struct channel *channel,
+		      const struct pubkey *side_payment_key,
+		      const struct pubkey *other_payment_key,
+		      const struct pubkey *side_revocation_key,
+		      const struct pubkey *side_delayed_payment_key,
+		      enum side side)
 {
-	struct bitcoin_tx *tx;
+	size_t i, n;
+	struct sha256_double txid;
+	u32 feerate_per_kw = channel->view[side].feerate_per_kw;
+
+	/* Get txid of commitment transaction */
+	bitcoin_txid((*txs)[0], &txid);
+
+	for (i = 0; i < tal_count(htlcmap); i++) {
+		const struct htlc *htlc = htlcmap[i];
+		struct bitcoin_tx *tx;
+		u8 *wscript;
+
+		if (!htlc)
```

Comment:
Yes, this is kind of confusing.  htlcmap contains NULLs where the to-local and to-remote outputs are (it's set by channel_tx()).  I'm not entirely happy with this, as it still doesn't tell us which of those outputs is which (we'll want this later, I'm pretty sure).